airsonic-main/src/main/java/org/airsonic/player/Application.java

```diff
@@ -162,6 +165,11 @@ public Filter noCacheFilter() {
     }
 
     private static SpringApplicationBuilder doConfigure(SpringApplicationBuilder application) {
+        // Handle HSQLDB database upgrades from 1.8 to 2.x before any beans are started.
+        application.application().addListeners((ApplicationListener<ApplicationEnvironmentPreparedEvent>) event -> {
+            LegacyHsqlUtil.upgradeHsqldbDatabaseSafely();
+        });
```

Comment:
Wouldn't users on a servlet container (Tomcat) miss this if this is not put inside a listener (such as in the main `main` method)?

(Also, this is minor but I like my logs to honor settings and be consistent, and putting this in the `main` method prevented logs from appearing as configured by Spring.)

Good idea about using the profile though, I'll add that.

Reply:
...shoot, it's not available in Spring Boot 1.5. Well, let's add a todo.

Reply:
normally yes, but the method isn't just called from `main()`

`doConfigure()` is also called here: https://github.com/airsonic/airsonic/pull/1345/files#diff-f4ea9b91519ec33fc48ca468f4d855fdR180-R182 which i think will ensure that the servlet container runs it. Best to try compiling and running it on a container, and see if stuff backs up

Reply:
Hum, I think I will just leave it as is. Putting it on application initialization seems like a natural place to me, with the added benefit that all the low-level Spring machinery is already initialized (so, less chance of being in a weird state).

Reply:
Okay, just note here then that at this point, for this event, the application has started but none of the initializers have run (including and especially CustomPropertySourceConfigurer), so no properties from airsonic or profiles can be read.

Also, without any checks, this means that you will touch and back up a db at the default location even if it's not being used. Personally, I would expect a program not to touch a location if I'm specifying a completely different location. For instance, this scenario will occur:

- Person1 specifies a url to connect at /var/myactualdb ("embed" i guess)
- Airsonic starts and updates /var/db instead and makes backup even though as a user I shouldn't really be expecting Airsonic to be touching that location
- (additionally it also connects and updates /var/myactualdb, but makes no backups of it)

Reply:
Ah, this scenario is a good point. I took some additional time to research and found a working solution that should work on both Spring 1.5 and 2 so that we only do the backups for the `legacy` profile. Let me know what you think!